compiler/qsc_qasm3/src/compiler.rs

```diff
@@ -1136,22 +1287,22 @@ impl QasmCompiler {
             &Type::Angle(..) => {
                 let msg = "Cast bit to angle";
                 self.push_unimplemented_error_message(msg, expr.span);
```

Comment:
Given that the calling code now pushes a semantic error when ::ExprKind::Err is returned, we are pushing both unimplemented and unsupported errors for the same cases.